src/providers/splash.ts

```diff
@@ -2,51 +2,67 @@ import { AmountWithDecimal, Provider } from "@/types/core";
 import { AmmPool, makeNativePools, Pools } from "@ergolabs/ergo-dex-sdk";
 import { AssetAmount, Explorer, RustModule } from "@ergolabs/ergo-sdk";
 
+type Order = {
+  price: string;
+  avgPrice: string;
+  ordersLiquidity: string;
+  poolsLiquidity: string;
+  accumulatedLiquidity: string;
+  volumeRelation: string;
+};
+
+type Pair = {
+  base: string;
+  quote: string;
+};
+
+type OrderBookResponse = {
+  pair: Pair;
+  asks: Order[];
+  bids: Order[];
+  spot: string;
+  previousSpot: string;
+  ammTotalLiquidityBase: string;
+  ammTotalLiquidityQuote: string;
+};
+
 class Splash implements Provider {
   private baseUrl: string;
 
   constructor(url: string) {
     this.baseUrl = url;
   }
 
-  async getOrderBook(marketId: string) {
-    try {
-      const response = await fetch(
-        `${this.baseUrl}/platform-api/v1/trading-view/order-book?base=${marketId}&quote=.`,
-        {
-          method: "GET",
-          headers: {
-            "Content-Type": "application/json",
-          },
-        }
-      );
-      return response.json();
-    } catch (error) {
-      console.error("Error fetching order book:", error);
-      throw new Error("Failed to fetch order book");
-    }
+  async getOrderBook(marketId: string): Promise<OrderBookResponse> {
+    const response = await fetch(
+      `${this.baseUrl}/platform-api/v1/trading-view/order-book?base=${marketId}&quote=.`,
+      {
+        method: "GET",
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    const data = await response.json();
+    return data as OrderBookResponse;
   }
 
   async x2y(marketId: string, amount: AmountWithDecimal) {
```

Comment:
As I mentioned, `amount` is of type `number`, and not `AmountWithDecimal` anymore. Make sure function parameters and return types match those of `Provider` interface itself.